Enable -Wvla when building WebKit

ddkilzer opened the pull request:
#### b39622eb0b6b8c1d0b248d189986100521b139e7
<pre>
Enable -Wvla when building WebKit
&lt;<a href="https://bugs.webkit.org/show_bug.cgi?id=275081">https://bugs.webkit.org/show_bug.cgi?id=275081</a>&gt;
&lt;<a href="https://rdar.apple.com/129190911">rdar://129190911</a>&gt;

Reviewed by Yusuke Suzuki.

* Configurations/CommonBase.xcconfig:
(WK_COMMON_WARNING_CFLAGS): Add -Wvla.
* Source/JavaScriptCore/API/tests/testapi.c:
(checkJSStringOOBUTF8):
(checkJSStringOOBUTF16):
(checkJSStringOOBUTF16AtEnd):
- Stop using VLAs.  Since this is C, we use malloc()/free() instead.
  Regressed recently in 279673@main.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
(WK_FIXME_WARNING_CFLAGS): Add -Wno-vla.
- The opus project and one pffft source file use VLAs.
* Tools/DumpRenderTree/cg/PixelDumpSupportCG.cpp:
(computeSHA1HashStringForBitmapContext):
- Use a Vector&lt;uint32_t&gt; instead of a VLA.
* Tools/DumpRenderTree/mac/AccessibilityNotificationHandler.mm:
(makeArrayRefForArray):
- Ignore -Wvla warnings for JSValueRef objects that must be allocated on
  the stack.
* Tools/WebKitTestRunner/InjectedBundle/cocoa/AccessibilityCommonCocoa.mm:
(WTR::makeJSArray): Ditto.
* Tools/WebKitTestRunner/ios/HIDEventGenerator.mm:
(-[HIDEventGenerator touchDown:touchCount:]):
(-[HIDEventGenerator liftUp:touchCount:]):
(-[HIDEventGenerator moveToPoints:touchCount:duration:]):
- Use a Vector&lt;CGPoint&gt; instead of a VLA.

Canonical link: <a href="https://commits.webkit.org/279702@main">https://commits.webkit.org/279702@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0c56ff493f1bda3848b85005f24283429bdc8793

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54210 "6 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33598 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6752 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57487 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4936 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56513 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41125 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4883 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43912 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3308 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56306 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/31860 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46949 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25054 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28670 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4275 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3083 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/47570 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50370 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4482 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59079 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/53718 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29416 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/4640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51336 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30588 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47057 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50696 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31562 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/66019 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8033 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30371 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12572 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Source/JavaScriptCore/API/tests/testapi.c:
```diff
@@ -1195,13 +1195,13 @@ static void checkJSStringOOBUTF8(void)
-    char outCString[outCStringSize + sourceCStringSize];
-    memset(outCString, 0x13, sizeof(outCString));
+    char *outCString = (char*)malloc(outCStringSize + sourceCStringSize);
+    memset(outCString, 0x13, sizeof(outCStringSize + sourceCStringSize));
```
Should this just be `memset(outCString, 0x13, outCStringSize + sourceCStringSize)`?

ddkilzer replied:
Yep!  Copy-pasta error.

ddkilzer replied:
Good catch!

Constellation commented on Tools/WebKitTestRunner/InjectedBundle/cocoa/AccessibilityCommonCocoa.mm:
```diff
@@ -66,10 +66,10 @@ Class webAccessibilityObjectWrapperClass()
 {
     NSUInteger count = array.count;
-    JSValueRef arguments[count];
+    Vector<JSValueRef> arguments(count);
```
JSValueRef is GC managed, so it needs to be stored on stack. So using heap Vector will cause GC crash.

Constellation commented on Source/JavaScriptCore/API/tests/testapi.c:
```diff
@@ -1195,13 +1195,13 @@ static void checkJSStringOOBUTF8(void)
 
-    char sourceCString[sourceCStringSize];
-    memset(sourceCString, 0, sizeof(sourceCString));
+    char *sourceCString = (char*)malloc(sourceCStringSize);
```
We should put `*` to `char` side since this is `C` file and it is C pointer (not Objective-C pointers).

Constellation commented on Source/JavaScriptCore/API/tests/testapi.c:
```diff
@@ -1195,13 +1195,13 @@ static void checkJSStringOOBUTF8(void)
 
-    char outCString[outCStringSize + sourceCStringSize];
-    memset(outCString, 0x13, sizeof(outCString));
+    char *outCString = (char*)malloc(outCStringSize + sourceCStringSize);
```
Ditto

Constellation commented on Source/JavaScriptCore/API/tests/testapi.c:
```diff
@@ -1218,15 +1218,17 @@ static void checkJSStringOOBUTF8(void)
 
-    char sourceCString[sourceCStringSize];
-    memset(sourceCString, 0, sizeof(sourceCString));
+    char *sourceCString = (char*)malloc(sourceCStringSize);
```
Ditto.

Constellation commented on Source/JavaScriptCore/API/tests/testapi.c:
```diff
@@ -1235,8 +1237,8 @@ static void checkJSStringOOBUTF16(void)
 
-    char outCString[outCStringSize + sourceCStringSize];
-    memset(outCString, 0x13, sizeof(outCString));
+    char *outCString = (char*)malloc(outCStringSize + sourceCStringSize);
```
Ditto.

Constellation commented on Source/JavaScriptCore/API/tests/testapi.c:
```diff
@@ -1253,15 +1255,17 @@ static void checkJSStringOOBUTF16(void)
 
-    char sourceCString[sourceCStringSize];
-    memset(sourceCString, 0, sizeof(sourceCString));
+    char *sourceCString = (char*)malloc(sourceCStringSize);
```
Ditto.

Constellation commented on Source/JavaScriptCore/API/tests/testapi.c:
```diff
@@ -1270,8 +1274,8 @@ static void checkJSStringOOBUTF16AtEnd(void)
 
-    char outCString[outCStringSize + sourceCStringSize];
-    memset(outCString, 0x13, sizeof(outCString));
+    char *outCString = (char*)malloc(outCStringSize + sourceCStringSize);
```
Ditto.

Constellation reviewed (approved):
r=me

Commit:
Enable -Wvla when building WebKit
<https://bugs.webkit.org/show_bug.cgi?id=275081>
<rdar://129190911>

Reviewed by Yusuke Suzuki.

* Configurations/CommonBase.xcconfig:
(WK_COMMON_WARNING_CFLAGS): Add -Wvla.
* Source/JavaScriptCore/API/tests/testapi.c:
(checkJSStringOOBUTF8):
(checkJSStringOOBUTF16):
(checkJSStringOOBUTF16AtEnd):
- Stop using VLAs.  Since this is C, we use malloc()/free() instead.
  Regressed recently in 279673@main.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
(WK_FIXME_WARNING_CFLAGS): Add -Wno-vla.
- The opus project and one pffft source file use VLAs.
* Tools/DumpRenderTree/cg/PixelDumpSupportCG.cpp:
(computeSHA1HashStringForBitmapContext):
- Use a Vector<uint32_t> instead of a VLA.
* Tools/DumpRenderTree/mac/AccessibilityNotificationHandler.mm:
(makeArrayRefForArray):
- Ignore -Wvla warnings for JSValueRef objects that must be allocated on
  the stack.
* Tools/WebKitTestRunner/InjectedBundle/cocoa/AccessibilityCommonCocoa.mm:
(WTR::makeJSArray): Ditto.
* Tools/WebKitTestRunner/ios/HIDEventGenerator.mm:
(-[HIDEventGenerator touchDown:touchCount:]):
(-[HIDEventGenerator liftUp:touchCount:]):
(-[HIDEventGenerator moveToPoints:touchCount:duration:]):
- Use a Vector<CGPoint> instead of a VLA.

Canonical link: https://commits.webkit.org/279702@main
